src/herder/test/HerderTests.cpp

```diff
@@ -5856,13 +5856,12 @@ TEST_CASE("exclude transactions by operation type", "[herder]")
 // ledger `n-1` when closing ledger `n`
 TEST_CASE("SCP message capture from previous ledger", "[herder]")
 {
-    constexpr uint32_t version =
-        static_cast<uint32_t>(SOROBAN_PROTOCOL_VERSION);
+    uint32_t version = static_cast<uint32_t>(SOROBAN_PROTOCOL_VERSION);
 
     // Initialize simulation
     auto networkID = sha256(getTestConfig().NETWORK_PASSPHRASE);
     auto simulation = std::make_shared<Simulation>(
-        Simulation::OVER_LOOPBACK, networkID, [](int i) {
+        Simulation::OVER_LOOPBACK, networkID, [version](int i) {
             auto cfg = getTestConfig(i, Config::TESTDB_ON_DISK_SQLITE);
             cfg.TESTING_UPGRADE_LEDGER_PROTOCOL_VERSION = version;
```

Comment:
is this needed at all? I think this test should just use the most recent version for testing, which is the default, if we don't override `TESTING_UPGRADE_LEDGER_PROTOCOL_VERSION`

Reply:
I have no idea why the test is hard-wired to a single version. possibly copypasta. it works fine without that line and in fact without the ConfigGen argument at all.